integration-test/helpers/pages/genericPage.js

```diff
@@ -10,15 +10,19 @@ import {constants} from '../mocks.js'
 export function genericPage (page) {
     class GenericPage {
         async passwordFieldShowsFillKey (selector = '#password') {
-            // don't make assertions until the element is both found + has a none-empty 'style' attribute
-            await page.waitForFunction(selector => Boolean(document.querySelector(selector)?.getAttribute('style')), selector)
+            // don't make assertions until the element has been scanned
+            const field = page.locator(selector)
+            await expect(field).toHaveAttribute('data-ddg-inputtype')
```

Comment:
I've switched this to use the locator with `expect` so that playwright automatically pierces the shadow tree and waits for the attribute to be set. Otherwise, I'd have to implement something custom for `waitForFunction`. This also seems much cleaner.

Reply:
👌